utils/mail.go

```diff
@@ -20,6 +21,32 @@ import (
 	"github.com/mattermost/mattermost-server/model"
 )
 
+type loginAuth struct {
+	username, password string
+}
+
+func LoginAuth(username, password string) smtp.Auth {
+	return &loginAuth{username, password}
+}
+
+func (a *loginAuth) Start(server *smtp.ServerInfo) (string, []byte, error) {
```

Comment:
I assume you'll inspect `server.Auth` so we can continue to use PLAIN where LOGIN isn't supported?

We should also continue to error out on `!server.TLS` for remote servers.

Reply:
ok sure!